crates/ruff/src/autofix/mod.rs

```diff
@@ -62,7 +62,7 @@ fn apply_fixes<'a>(
             continue;
         }
 
-        for edit in fix.edits() {
+        for edit in fix.edits().iter().sorted_unstable_by_key(|e| e.start()) {
```

Comment:
For my own curiosity, why do we need `sorted_unstable` here?

Reply:
Why `sort_unstable` over `sort`. IIUC, this PR basically sorts fixes to avoid overlaps. From a (brief) reading, `unstable_sort` doesn't preserve order between equal-ordered components, which doesn't seem like it matters for this PR. So I'm wondering, if it doesn't matter, why we're specifying `unstable`. 

Obviously I could very well be wrong that it doesn't matter as well :)

Reply:
The main motivation is that `sort_unstable` tends to be faster than `sort` and two edits in the same `Fix` with the same start position are problematic anyway (should really be a single edit).

> It is typically faster than stable sorting, except in a few special cases, e.g., when the slice consists of several concatenated sorted sequences. [source](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable)